Source/Thunder/PluginServer.cpp

```diff
@@ -1224,6 +1227,30 @@ namespace PluginHost {
         }
     }
 
+    void Server::StateControlStateChange(const string& callsign, const IStateControl::state state)
+    {
+        ASSERT((_controller.IsValid() == true) && (_controller->ClassType<Plugin::Controller>() != nullptr));
+        Plugin::Controller* controller = _controller->ClassType<Plugin::Controller>();
+        ASSERT(controller != nullptr);
+
+        switch (state) {
+            case IStateControl::state::SUSPENDED:
+                controller->NotifyStateControlStateChange(callsign, Exchange::Controller::ILifeTime::state::SUSPENDED);
+                break;
+            case IStateControl::state::RESUMED:
+                controller->NotifyStateControlStateChange(callsign, Exchange::Controller::ILifeTime::state::RESUMED);
+                break;
+            case IStateControl::state::EXITED:
```

Comment:
Just checked the EXITED state, since I did not understaood what it meant, and it looks like it is not used at all. Could we try to remove it (from interface and here) and see if it still all compiles ?

Reply:
It is used, if the plugin/application itself requests to quit.